src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp

```diff
@@ -680,29 +723,27 @@ bool DiscoveryDataBase::process_dirty_topics()
 bool DiscoveryDataBase::delete_entity_of_change(
         fastrtps::rtps::CacheChange_t* change)
 {
-    (void)change;
-    return true;
-    /*
-       if (change->kind != fastrtps::rtps::ChangeKind_t::ALIVE)
-       {
-        logWarning(DISCOVERY_DATABASE, "Attempting to delete information of an ALIVE entity: " << guid_from_change_(change));
+
+    if (change->kind != fastrtps::rtps::ChangeKind_t::ALIVE)
```

Comment:
According to the logWarning it should be like this:

```suggestion
    if (change->kind == fastrtps::rtps::ChangeKind_t::ALIVE)
```